### PR TITLE
ci: bump GH Actions to Ubuntu Jammy where applicable

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
This should be ready to go. There are two "issues" which need to be addressed later though:

1) The FIXME introduced by the last commit - I don't particularly like this workaround, but I don't know about a better way, since we can't tell the actionlint to ignore the errors due to https://github.com/rhysd/actionlint/blob/main/docs/usage.md#super-linter. I'll keep track of https://github.com/github/super-linter/pull/2897 and once it's merged I'll revert the workaround.

2) TODO in the mkosi action:
https://github.com/systemd/systemd/blob/1f9aa3d2a654531df10869aee324a687fbe5179f/.github/workflows/mkosi.yml#L83-L85

This apparently needs to wait until Ubuntu Jammy gets out of the public beta in Actions, since the mkosi action's cache currently causes issues: https://github.com/systemd/systemd/runs/6537391679?check_suite_focus=true#step:3:862 (/cc @DaanDeMeyer)

**Comment by mrc0mmand** [comment]
>     2. TODO in the mkosi action:
>        https://github.com/systemd/systemd/blob/1f9aa3d2a654531df10869aee324a687fbe5179f/.github/workflows/mkosi.yml#L83-L85
> 
> 
> This apparently needs to wait until Ubuntu Jammy gets out of the public beta in Actions, since the mkosi action's cache currently causes issues: https://github.com/systemd/systemd/runs/6537391679?check_suite_focus=true#step:3:862 (/cc @DaanDeMeyer)

Ad this issue - I think the root cause is that `runner.os` yields only name of the OS[0], not its version, hence the same cache is used across all supported Ubuntu versions[1]. I'm not really sure what would be the best solution here.

[0] https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context
[1] https://github.com/systemd/mkosi/blob/0adce3d326951b24d9e2ef808f9c3a4b52a936a7/action.yaml#L9-L12

File: src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c [hunk 1]
```diff
@@ -39,7 +39,7 @@ _public_ int cryptsetup_token_open_pin(
 
         /* This must not fail at this moment (internal error) */
         r = crypt_token_json_get(cd, token, &json);
-        assert(token == r);
+        assert_se(token == r);
```
Comment (yuwata) [inline src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:43]: Could you elaborate this change?

Comment (mrc0mmand) [reply src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:43]: Ah, I forgot to add the error to the commit message, my apologies:

```
2022-05-21T15:25:06.8452819Z FAILED: libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o 
2022-05-21T15:25:06.8457199Z clang-13 -Ilibcryptsetup-token-systemd-fido2.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o -MF libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o.d -o libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c
2022-05-21T15:25:06.8460369Z ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:33:13: error: variable 'r' set but not used [-Werror,-Wunused-but-set-variable]
2022-05-21T15:25:06.8460944Z         int r;
2022-05-21T15:25:06.8461146Z             ^
2022-05-21T15:25:06.8461321Z 1 error generated.
2022-05-21T15:25:06.9587264Z [516/2143] clang-13 -Ilibcryptsetup-token-systemd-pkcs11.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -I/usr/include/p11-kit-1 -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -MF libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o.d -o libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c
2022-05-21T15:25:06.9590826Z FAILED: libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o 
2022-05-21T15:25:06.9595399Z clang-13 -Ilibcryptsetup-token-systemd-pkcs11.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -I/usr/include/p11-kit-1 -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -MF libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o.d -o libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c
2022-05-21T15:25:06.9646280Z ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c:34:13: error: variable 'r' set but not used [-Werror,-Wunused-but-set-variable]
2022-05-21T15:25:06.9646679Z         int r;
2022-05-21T15:25:06.9646856Z             ^
2022-05-21T15:25:06.9647035Z 1 error generated.
2022-05-21T15:25:07.0143564Z [517/2143] /home/runner/.local/bin/meson --internal exe --capture src/core/bpf/socket_bind/socket-bind.skel.h -- /usr/sbin/bpftool gen skeleton src/core/bpf/socket_bind/socket-bind.bpf.o
2022-05-21T15:25:07.0144414Z ninja: build stopped: subcommand failed.
2022-05-21T15:25:07.0656402Z + fatal ''\''meson compile'\'' failed with -Db_ndebug=true'
```

Comment (yuwata) [reply src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:43]: I see. Then, could you add a short description why we need to use `assert_se()` here?

Comment (mrc0mmand) [reply src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:43]: Will do, together with updating the commit message.

**Review by yuwata — APPROVED** [review]

**Review by bluca — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
@mrc0mmand I'm working on getting mkosi on 22.04 in https://github.com/systemd/mkosi/pull/974

File: .github/workflows/coverity.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ permissions:
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (evverx) [inline .github/workflows/coverity.yml:17]: Nice! Now that all the dependencies are new enough both Coverity and CodeQL cover all of them finally.

File: .github/workflows/build_test.sh [hunk 3]
```diff
@@ -76,17 +76,16 @@ if [[ "$COMPILER" == clang ]]; then
...
     if ! apt install --dry-run "llvm-$COMPILER_VERSION" >/dev/null; then
         # Latest LLVM stack deb packages provided by https://apt.llvm.org/
         # Following snippet was partly borrowed from https://apt.llvm.org/llvm.sh
         wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --yes --dearmor --output /usr/share/keyrings/apt-llvm-org.gpg
         printf "deb [signed-by=/usr/share/keyrings/apt-llvm-org.gpg] http://apt.llvm.org/%s/   llvm-toolchain-%s-%s  main\n" \
                "$RELEASE" "$RELEASE" "$COMPILER_VERSION" >/etc/apt/sources.list.d/llvm-toolchain.list
-        PACKAGES+=("clang-$COMPILER_VERSION" "lldb-$COMPILER_VERSION" "lld-$COMPILER_VERSION" "clangd-$COMPILER_VERSION")
     fi
+
+    PACKAGES+=("clang-$COMPILER_VERSION" "lldb-$COMPILER_VERSION" "lld-$COMPILER_VERSION" "clangd-$COMPILER_VERSION")
```
Comment (evverx) [inline .github/workflows/build_test.sh:88]: @mrc0mmand I've just opened https://github.com/systemd/systemd/pull/23482 where I bumped `gcc` and changed the "gcc" stage accordingly.

**Pull request merged**